src/lib/deploy.js

```diff
@@ -344,20 +345,17 @@ async function deploy({ alias, yes }) {
   let transaction = await step('Build transaction', async () => {
     let Network = Mina.Network(graphQLEndpoint);
     Mina.setActiveInstance(Network);
-    let tx = await Mina.transaction(
-      { feePayerKey: zkAppPrivateKey, fee },
-      () => {
-        let zkapp = new zkApp(zkAppAddress);
-        zkapp.deploy({ verificationKey, zkappKey: zkAppPrivateKey });
-        // manually patch the tx (TODO: have this implemented properly in snarkyjs)
-        // remove the nonce precondition
-        zkapp.self.body.preconditions.account.nonce.isSome = Bool(false);
-        // don't increment the nonce
-        zkapp.self.body.incrementNonce = Bool(false);
-        // use full commitment (means with include the fee payer in the signature, so we're protected against replays)
-        zkapp.self.body.useFullCommitment = Bool(true);
-      }
-    );
+    let tx = await Mina.transaction({ sender: zkAppAddress, fee }, () => {
+      let zkapp = new zkApp(zkAppAddress);
+      zkapp.deploy({ verificationKey, zkappKey: zkAppPrivateKey });
+      // manually patch the tx (TODO: have this implemented properly in snarkyjs)
+      // remove the nonce precondition
+      zkapp.self.body.preconditions.account.nonce.isSome = Bool(false);
+      // don't increment the nonce
+      zkapp.self.body.incrementNonce = Bool(false);
+      // use full commitment (means with include the fee payer in the signature, so we're protected against replays)
+      zkapp.self.body.useFullCommitment = Bool(true);
+    });
     return { tx, json: tx.sign().toJSON() };
```

Comment:
since we removed the private key here, I think we have to add it when signing later: `tx.sign([zkAppPrivateKey]).toJSON()`

Reply:
Good catch.